tests/trainer/test_trainer.py

```diff
@@ -641,12 +641,15 @@ def test_trainer_max_steps_accumulate_batches(tmpdir):
 @pytest.mark.parametrize(
     ["benchmark_", "deterministic", "expected"],
     [
-        (None, False, True),
+        (None, False, None),
         (None, True, False),
+        (None, None, None),
```

Comment:
So now, somebody that creates `Trainer()` with no arguments, won't get `cudnn.benchmark=True` as it does in master?

I think having `benchmark=True` by default is best.

The proposal in this PR only works if we were able to know if the user had set `benchmark` manually before.

Reply:
The justification for this change is that Pytorch Lightning shouldn't break standard Pytorch functionality, but extend it. In general, PL offers value by establishing sensible default values so users don't need to think about them usually. However, in this case, we are silently setting a global variable, resulting in this behaviour:

```python
import torch
from pytorch_lightning import Trainer

torch.backends.cudnn.benchmark = True
trainer = Trainer(gpus=1)
print(torch.backends.cudnn.benchmark)
```

Output:

```
False
# When it should be True
```

Reply:
I understand that, but since there's no way for us to know whether the user previously set the value, we have 2 exclusive options:
1. Have the better default for most people which may override an existing value (current master)
2. Always respect the existing value but users have to remember to set this flag (this PR)

I personally prefer 1 as it establishes a "sensible default". We could request the torch folks to add an optional default so the options are not exclusive in the future.

ccing revierwers @awaelchli @ananthsub @krshrimali to see if they think differently.

Reply:
> I understand that, but since there's no way for us to know whether the user previously set the value,

We can know, by making the default value for the argument None. A default `Trainer()` would result in taking the value from `torch.backends.cudnn.benchmark` which by default is True

Reply:
> torch.backends.cudnn.benchmark which by default is True

Is it? locally:

```bash
$ python -c 'import torch; print(torch.backends.cudnn.benchmark)'
False
```